woocommerce/woocommerce-android
# Add a card reader config which has all the necessary details related to specific country

Closes: #5703 


### Description

This PR adds a card reader config that has all the necessary details for the card reader related to the specific country. 

The need for something like this was born out of the necessity that for In-Person Payments Interac transactions that happen for Canada, [we need to pass payment method type as `interac_present`](https://stripe.com/docs/terminal/payments/regional?integration-country=CA#create-a-paymentintent) along with the `card_present`. 

Whereas, for the US the payment method type must just be `card_present`. Also, Now that we support multiple countries for IPP. We need to verify the currency, supported readers...etc. 

Some config specific to countries like IPP in Canada won't support Stripe Extension for the first iteration whereas Stripe Extension is supported in the US, Should the reader capture the payment in the backend or in the app...etc goes into this configuration class as well.

We needed a place that has all the necessary config details related to the card reader specific to the country it is registered on so that we can verify anything we need with that one instance.

### Testing instructions

1. Make sure CI is happy/green
2. Verify IPP works end to end. 
    1.  Navigate to any store eligible for IPP
    2. Navigate to any order eligible for IPP
    3. Tap the `collect payment` button
    4. Make sure the payment succeeds or fails for a valid reason
    5. Make sure there is nothing weird happening during the payment flow


<br />
